src/DeviceInterface/WindowsOverlayDeviceInterface.js

```diff
@@ -0,0 +1,59 @@
+import InterfacePrototype from './InterfacePrototype.js'
+import {HTMLTooltipUIController} from '../UI/controllers/HTMLTooltipUIController.js'
+import {GetAutofillInitDataCall, SetSizeCall} from '../deviceApiCalls/__generated__/deviceApiCalls.js'
+import {overlayApi} from './overlayApi.js'
+
+/**
+ * This subclass is designed to separate code that *only* runs inside the
+ * Windows Overlay into a single place.
+ *
+ * It has some subtle differences to the macOS version, which is why
+ * this is another DeviceInterface
+ */
+export class WindowsOverlayDeviceInterface extends InterfacePrototype {
+    /**
+     * Mark top frame as not stripping credential data
+     * @type {boolean}
+     */
+    stripCredentials = false;
+
+    /**
+     * Because we're running inside the Overlay, we always create the HTML
+     * Tooltip controller.
+     *
+     * @override
+     * @returns {import("../UI/controllers/UIController.js").UIController}
+     */
+    createUIController () {
+        return new HTMLTooltipUIController({
+            tooltipKind: /** @type {const} */ ('modern'),
+            device: this
+        }, {
+            wrapperClass: 'top-autofill',
+            tooltipPositionClass: () => '.wrapper { transform: none; }',
+            setSize: (details) => this.deviceApi.notify(new SetSizeCall(details)),
+            testMode: this.isTestMode(),
+            /**
+             * Note: This is needed because Mutation observer didn't support visibility checks on Windows
```

Comment:
I think only Chrome supports `isVisible`, that's why we have `if (typeof change.isVisible === 'undefined')`. Is this needed because on Windows it is _not_ `undefined` and also _not_ working? 🙃 Can we add a different check right in the `safeExecute` function instead of this option here? It seems like an API limitation of the platform rather than an option we should handle ourselves. If the platform does implement `IntersectionObserver` v2 we want it to work as expected out of the box, rather than hunting down this option.

Reply:
I can look into this - we'd need to determine a safe alternative, and then test it out thoroughly on Windows 👍🏻